Sources/ApodiniDatabase/PropertyVisitors/VisitableFieldProperty.swift

```diff
@@ -0,0 +1,57 @@
+import Fluent
+
+/// A protocol to make the `FieldProperty` of a database model accessible at run time.
+/// It returns an `AnyCodable` object which contains the value associated with this `FieldProperty`
+protocol VisitableFieldProperty {
+    func accept<Visitor: FieldPropertyVisitor>(_ visitor: Visitor) -> Visitor.Value where Visitor.Value == TypeContainer
+}
+
+/// A protocol to make the `FieldProperty` of a database model accessible at run time.
+/// It returns an `AnyCodable` object which contains the value associated with this `IDProperty`
+protocol VisitableIDProperty {
+    func accept<Visitor: IDPropertyVisitor>(_ visitor: Visitor) -> Visitor.Value where Visitor.Value == TypeContainer
+}
+
+/// A protocol all field property visitors have to conform to. It returns whatever has been specified as `Value`.
+protocol FieldPropertyVisitor where Value: Codable {
+    associatedtype Value
```

Comment:
The `associatedtype` is not needed in the current implementation as far as I can see as you only use the concrete type `TypedContainer` everywhere. I would suggest removing it to reduce the complexity of reading the code and steering the type checker a bit less 😅